### DISCUSSION TITLE
Test dependencies for process serialization

### DISCUSSION THREAD

**Discussion by amikan** [avocado-framework/avocado#5423] [Q&A] [answered]
Hello, after getting legacy runner deprecated I have a question about how to achieve the same execution behavior with nrunner.
I have few tests to be executed after some other test finished. It was working good when using legacy runner, which executed tests one by one. Is there any way to disable parallel execution for nrunner?

Tried to implement the same with asset requirements field, but looks like it can't detect asset created after test starting.

Overall I'm little bit surprised there is an API for package or asset requirements, but there is no API to depend on other test execution result.

**Comment by richtja** [comment, marked as answer]
Hi @amikan for disabling the parallel execution you have to set `--nrunner-max-parallel-tasks=1` more info in [documentation](https://avocado-framework.readthedocs.io/en/latest/config/index.html#nrunner-max-parallel-tasks). Right now, Avocado doesn't support tests depending on each other, but we plan to do so in #5205.

**Reply by amikan** [reply]
Are there any plans for implementing 'test' dependency type mentioned in [BP004](https://github.com/avocado-framework/avocado/blob/master/docs/source/blueprints/BP004.rst) in [next version #101](https://github.com/avocado-framework/avocado/milestone/27)? I see 'Dependency Graph' related milestones but nothing about 'test' dependency itself.

We have a testsuite full of test cases of 2 different types:
```
A. 'Build' test case for some arch, which generates a runnable image for qemu.
B. 'Run' test case for the same arch, which executes qemu with the image prepared in A.
```
So our testsuite looks like:
```
1) build 'image1'
2) build 'image2'
3) build 'image3'
...
4) run 'image1'
5) run 'image2'
6) run 'image3'
...
```

We've fixed `max-parallel-tasks` to 1 currently to have all those test cases run one after another, but it will be much more optimal to set a dependency of 'test' type for 4-6 test cases and run all of them in parallel.
Moreover it looks like there is no objections in using that kind of dependency right now in such a testsuite, because we can depend one 'run' test case on one 'build' test case without any conflicts.
So 'test' dependency type seems to be a separate feature not really connected to [BP004](https://github.com/avocado-framework/avocado/blob/master/docs/source/blueprints/BP004.rst) implementation.

**Reply by richtja** [reply]
Hi @amikan we have issue #5205 about this topic, which should implement `test` dependencies, but unfortunately this feature is in our long term backlog and we don't plan to implementing it any time soon. I understand that it will be very useful for your use-case, but we don't have resources to implemented it now. But if you are interested in this feature than any contributions are very welcome and we will support you in this afford.

**Comment by amikan** [comment]
That works, thank you!

**Reply by marioaag** [reply]
Just an idea, that maybe could help to solve this is generate multiple test suits, if you pass all of them to the same job it will run one test suit after the other and all the jobs inside the same test suit will be run in parallel.
